rotkehlchen/exchanges/coinbaseprime.py

```diff
@@ -0,0 +1,458 @@
+    ) -> dict[str, Any]:
+        uri = f'{PRIME_BASE_URL}/{module}'
+        if path != '':
+            uri += f'/{path}'
+
+        url_path = urlparse(uri).path
+        timestamp = Timestamp(int(time.time()))
+        self.session.headers.update({
+            'X-CB-ACCESS-TIMESTAMP': str(timestamp),
+            'X-CB-ACCESS-SIGNATURE': self.sign(timestamp, url_path),
+        })
+        log.debug(f'Querying coinbase prime module {module}/{path} with {params=}')
+        try:
+            response = self.session.get(url=uri, params=params)
+        except requests.RequestException as e:
+            raise RemoteError(f'Coinbase Prime API request failed due to {e}') from e
+
+        try:
+            data = response.json()
+        except requests.JSONDecodeError as e:
+            raise RemoteError(f'Coinbase Prime returned invalid json {response.text}') from e
+
+        return data
+
+    @overload
+    def _query_paginated_endpoint(
+            self,
+            query_params: dict[str, Any],
+            portfolio_id: str,
+            method: Literal['transactions'],
+            decoding_logic: Callable[[dict[str, Any]], AssetMovement | None],
+    ) -> list[AssetMovement]:
+        ...
+
+    @overload
+    def _query_paginated_endpoint(
+            self,
+            query_params: dict[str, Any],
+            portfolio_id: str,
+            method: Literal['orders'],
+            decoding_logic: Callable[[dict[str, Any]], Trade | None],
+    ) -> list[Trade]:
+        ...
+
+    def _query_paginated_endpoint(
+            self,
+            query_params: dict[str, Any],
+            portfolio_id: str,
+            method: Literal['orders', 'transactions'],
+            decoding_logic: Callable[[dict[str, Any]], AssetMovement | None] | Callable[[dict[str, Any]], Trade | None],  # noqa: E501
+    ) -> list[Trade] | list[AssetMovement]:
+        """Abstraction to consume all the events in the selected queries.
+        It uses the `decoding_logic` to process the different events and returns a list
+        that depends on the signature of it.
+
+        This function may raise:
+        - RemoteError
+        """
+        result = []
+        while True:
+            response = self._api_query(
+                module='portfolios',
+                path=f'{portfolio_id}/{method}',
+                params=query_params,
+            )
+
+            events = response[method]
+            for raw_event in events:
+                try:
+                    if (event := decoding_logic(raw_event)) is None:
+                        log.warning(
+                            f'Wont process event {raw_event} from coinbase prime. Skipping',
+                        )
+                    else:
+                        result.append(event)
+                except DeserializationError as e:
+                    self.msg_aggregator.add_error(
+                        f'Failed to process coinbase prime trade due to {e}. Skipping entry...',
```

Comment:
Is it always trades here?

```suggestion
                        f'Failed to process coinbase prime event due to {e}. Skipping entry...',
```